eng/Packages.Data.props

```diff
@@ -55,7 +55,7 @@
     Only packages that are approved dependencies should go here.
   -->
 
-  <ItemGroup Condition="'$(IsClientLibrary)' == 'true'">
+  <ItemGroup Condition="'$(IsClientLibrary)' == 'true' or '$(IsTestProject)' == 'true'">
```

Comment:
Do we need this one? we have all test dependencies in it's own group.

Reply:
I wanted to avoid someone changing shipping package dependencies when trying to update a test dependency.

Reply:
I hit a case where a test project, that doesn't have IsClientLibrary set but still needs Azure.Identity as a dependency. Instead of duplicating Azure.Identity I decided to enable this for test projects also. We should definitely look closer at refactoring this file and even potentially splitting it up.

Reply:
Can we add Azure.Identity to a list of test packages instead?

Reply:
We could definitely do that as well, but it would be a duplicate entry with a duplicate version which isn't ideal but it would definitely solve this problem as well.

Reply:
We already have duplicated or different versions for many libraries. We can definitely revisit the design of the version list but I don't want to do it as part of this PR.